### DISCUSSION TITLE
Could we implement speaker diarization?

### DISCUSSION THREAD

**Discussion by DrMemoryFish** [openai/whisper#273] [Ideas]
I'm not experienced in discussions within Git (i don't even know if this is allowed), let alone knowledgeable when it comes to AI or ML, but could we implement speaker diarization within whisper? 

for example, you could have a built script. sometimes I'm in class and I want my voice notes to be transcribed in a structured format, especially in meetings and calls. or when rehearsing, it's nice to have the script as person 1, person 2 and so on, without manually editing the script. also for youtube subtitles, it's nice to know who is speaking.

what are people's thoughts on speaker diarization? I know it's possible to implement, but if it's not possible to implement (correct me if I'm wrong) why? and if not possible RIGHT NOW but CAN be implemented, how soon can we expect to see it inside whisper? will the be a whole separate repo for speaker diarization?

I'm not sure if I'm talking too much, so ill stop here. please do share your thoughts, criticism is welcome, please do forgive my grammar I'm not the best at writing my thought out like this, I hope you all understand what I'm saying though (if not please ask questions)

**Comment by Majdoddin** [comment]
Hi @DrMemoryFish, 
[www.lexicaps.com](http://www.lexicaps.com/) seamlessly adds diarization to Whispers transcription. No 3rd party packages.
Feel free to contact me if you need more features.
Announcement: https://github.com/openai/whisper/discussions/1537
Repo: https://github.com/Majdoddin/lexicaps
